VagrantProvision.sh

```diff
@@ -193,89 +254,90 @@ PATH=$HOOT_HOME/bin:$PATH
 fi
 
-# Allow linking in Tomcat context
 if ! grep -i --quiet 'allowLinking="true"' /etc/tomcat6/context.xml; then
-    echo "Set allowLinking to true in Tomcat context"
+    echo "Set allowLinking to true in Tomcat context..."
     sudo sed -i.bak "s@^<Context>@<Context allowLinking=\"true\">@" /etc/tomcat6/context.xml
 fi
 
-# Create directory for webapp
-if [ ! -d /usr/share/tomcat6/.deegree ]; then
-    echo "Creating directory for webapp"
-    sudo mkdir /usr/share/tomcat6/.deegree
-    sudo chown tomcat6:tomcat6 /usr/share/tomcat6/.deegree
-fi
-
-# Tweak dev environment to make tests run faster
-if [ ! -f $HOOT_HOME/hoot-services/src/main/resources/conf/local.conf ]; then
-    echo 'testJobStatusPollerTimeout=250' > $HOOT_HOME/hoot-services/src/main/resources/conf/local.conf
+if [ ! -d $TOMCAT6_HOME/.deegree ]; then
+    echo "Creating deegree directory for webapp..."
+    sudo mkdir $TOMCAT6_HOME/.deegree
+    sudo chown vagrant:tomcat6 $TOMCAT6_HOME/.deegree
 fi
 
-# Update the init.d script for node-mapnik-server
-sudo cp node-mapnik-server/init.d/node-mapnik-server /etc/init.d
-sudo chmod a+x /etc/init.d/node-mapnik-server
-# Make sure all npm modules are installed
-cd node-mapnik-server
-npm install
-cd ..
-
 # Update marker file date now that dependency and config stuff has run
 # The make command will exit and provide a warning to run 'vagrant provision'
 # if the marker file is older than this file (VagrantProvision.sh)
 touch Vagrant.marker
 
-# Build Hoot
-echo "Configuring Hoot"
-aclocal && autoconf && autoheader && automake && ./configure -q --with-rnd --with-services
+echo "Configuring Hoot..."
+echo HOOT_HOME: $HOOT_HOME
+cp conf/DatabaseConfig.sh.orig conf/DatabaseConfig.sh
+
+if [ ! -f /etc/init.d/node-mapnik-server ]; then
+  echo "Installing node-mapnik-server..."
+  sudo cp $HOOT_HOME/node-mapnik-server/init.d/node-mapnik-server /etc/init.d
+  sudo chmod a+x /etc/init.d/node-mapnik-server
+  # Make sure all npm modules are installed
+  cd node-mapnik-server
+  sudo npm install
+  cd ..
+fi
+
+aclocal && autoconf && autoheader && automake && ./configure --with-rnd --with-services --with-uitests
 if [ ! -f LocalConfig.pri ] && ! grep --quiet QMAKE_CXX LocalConfig.pri; then
-    echo 'Customizing LocalConfig.pri'
+    echo 'Customizing LocalConfig.pri...'
     cp LocalConfig.pri.orig LocalConfig.pri
     echo 'QMAKE_CXX=ccache g++' >> LocalConfig.pri
 fi
-echo "Building Hoot"
-make clean -sj$(nproc)
+echo "Building Hoot... "
+echo "Will take several extra minutes to build the training data the initial time Hootenanny is installed only."
+# make clean-all -sj$(nproc)
+# The services build won't always complete the first time without errors (for some unknown reason caused by 
+# the Maven pom), so we're executing multiple compiles here to get around that.
+make -sj$(nproc) &> /dev/null || true
```

Comment:
Is it still an issue? It used to be a problem many months ago, but I haven't seen it lately.